modules/likes/queuehandler.js

```diff
@@ -213,66 +223,155 @@ jQuery( document ).click( function( e ) {
-	} else {
-		$wrapper.find( '.post-likes-widget-placeholder' ).after( '<iframe class="post-likes-widget jetpack-likes-widget" name="' + $wrapper.data( 'name' ) + '" height="55px" width="100%" frameBorder="0" src="' + $wrapper.data( 'src' ) + '"></iframe>' );
+	var placeholder = $wrapper.find( '.likes-widget-placeholder' );
+
+	// Post like iframe
+	if ( placeholder.hasClass( 'post-likes-widget-placeholder' ) ) {
+		var postLikesFrame = document.createElement( 'iframe' );
+
+		postLikesFrame['class'] = 'post-likes-widget jetpack-likes-widget';
+		postLikesFrame.name = $wrapper.data( 'name' );
+		postLikesFrame.src = $wrapper.data( 'src' );
+		postLikesFrame.height = '18px';
+		postLikesFrame.width = '200px';
+		postLikesFrame.frameBorder = '0';
+		postLikesFrame.scrolling = 'no';
+
+		if ( $wrapper.hasClass( 'slim-likes-widget' ) ) {
+			postLikesFrame.height = '22px';
+			postLikesFrame.width = '68px';
+			postLikesFrame.scrolling = 'no';
+		} else {
+			postLikesFrame.height = '55px';
+			postLikesFrame.width = '100%';
+		}
+
+		placeholder.after( postLikesFrame );
+	}
+
+	// Comment like iframe
+	if ( placeholder.hasClass( 'comment-likes-widget-placeholder' ) ) {
+		var commentLikesFrame = document.createElement( 'iframe' );
+
+		commentLikesFrame['class'] = 'comment-likes-widget-frame jetpack-likes-widget-frame';
+		commentLikesFrame.name = $wrapper.data( 'name' );
+		commentLikesFrame.src = $wrapper.data( 'src' );
+		commentLikesFrame.height = '18px';
+		commentLikesFrame.width = '200px';
+		commentLikesFrame.frameBorder = '0';
+		commentLikesFrame.scrolling = 'no';
+
+		$wrapper.find( '.comment-like-feedback' ).after( commentLikesFrame );
+		
+		jetpackCommentLikesLoadedWidgets.push( commentLikesFrame );
 	}
 
 	$wrapper.removeClass( 'jetpack-likes-widget-unloaded' ).addClass( 'jetpack-likes-widget-loading' );
 
 	$wrapper.find( 'iframe' ).load( function( e ) {
 		var $iframe = jQuery( e.target );
-		$wrapper.removeClass( 'jetpack-likes-widget-loading' ).addClass( 'jetpack-likes-widget-loaded' );
 
-		JetpackLikespostMessage( { event: 'loadLikeWidget', name: $iframe.attr( 'name' ), width: $iframe.width() }, window.frames[ 'likes-master' ] );
+		JetpackLikesPostMessage( { event: 'loadLikeWidget', name: $iframe.attr( 'name' ), width: $iframe.width() }, window.frames[ 'likes-master' ] );
+
+		$wrapper.removeClass( 'jetpack-likes-widget-loading' ).addClass( 'jetpack-likes-widget-loaded' );
 
 		if ( $wrapper.hasClass( 'slim-likes-widget' ) ) {
 			$wrapper.find( 'iframe' ).Jetpack( 'resizeable' );
 		}
 	});
-	setTimeout( JetpackLikesWidgetQueueHandler, 250 );
 }
+
+function jetpackGetUnloadedWidgetsInView() {
+	var $unloadedWidgets = jQuery( 'div.jetpack-likes-widget-unloaded' );
+
+	return $unloadedWidgets.filter( function() {
+		return jetpackIsScrolledIntoView( this );
+	} );
```

Comment:
I'm curious on the inconsistent use of the array functions. above we use a `for` loop but here a `filter`

support for map/filter/reduce/forEach comes in at IE9. if we are okay with that, it would be nice to explore using them above. if not, we should probably remove this one.